sql/parser/datum.go

```diff
@@ -237,7 +237,28 @@ func (d DFloat) IsMin() bool {
 }
 
 func (d DFloat) String() string {
-	return strconv.FormatFloat(float64(d), 'g', -1, 64)
+	s := strconv.FormatFloat(float64(d), 'g', -1, 64)
+	if parsesAsInt(s) {
```

Comment:
I like this approach, but it requires some knowledge of the stdlib's definition of 'large exponent', since large integers will pass the Modf test, but be printed with an exponent, and thus don't need the '.0'.
Example: https://play.golang.org/